subsys/nrf_security/src/drivers/cracen/cracenpsa/src/platform_keys/platform_keys.c

```diff
@@ -0,0 +1,291 @@
+				     PSA_KEY_PERSISTENCE_READ_ONLY, PSA_KEY_LOCATION_CRACEN));
+	psa_set_key_type(&mkek_attr, PSA_KEY_TYPE_AES);
+
+	cracen_aead_operation_t op = {};
+	psa_status_t status = cracen_aead_decrypt_setup(
+		&op, &mkek_attr, NULL, 0,
+		PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_GCM, key.mac_size));
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+
+	status = cracen_aead_set_nonce(&op, (uint8_t *)key.nonce, sizeof(key.nonce));
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+
+	status = cracen_aead_update_ad(&op, (uint8_t *)&key.type, sizeof(key.type));
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+	status = cracen_aead_update_ad(&op, (uint8_t *)&key.bits, sizeof(key.bits));
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+	status = cracen_aead_update_ad(&op, key_buffer, key_buffer_size);
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+
+	size_t outlen;
+
+	status = cracen_aead_finish(&op, NULL, 0, &outlen, key.mac, key.mac_size, &key.mac_size);
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+
+	psa_set_key_bits(attributes, key.bits);
+	psa_set_key_type(attributes, key.type);
+
+	if (key.type == PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_FAMILY_TWISTED_EDWARDS)) {
+		psa_set_key_algorithm(attributes, PSA_ALG_PURE_EDDSA);
+		psa_set_key_usage_flags(attributes, PSA_KEY_USAGE_VERIFY_MESSAGE);
+	} else {
+		return PSA_ERROR_INVALID_HANDLE;
+	}
+
+	size_t key_bytes = PSA_BITS_TO_BYTES(key.bits);
+
+	if (key_bytes < key_buffer_size) {
+		*key_buffer_length = 0;
+		return PSA_ERROR_BUFFER_TOO_SMALL;
+	}
+
+	memcpy(key_buffer, key.key_buffer, key_bytes);
+	*key_buffer_length = key_bytes;
+
+	return PSA_SUCCESS;
+}
+
+size_t cracen_platform_keys_get_size(psa_key_attributes_t const *attributes)
+{
+	sicr_key key = find_key(MBEDTLS_SVC_KEY_ID_GET_KEY_ID(psa_get_key_id(attributes)));
+
+	return PSA_BITS_TO_BYTES(key.bits);
+}
+
+psa_status_t cracen_platform_get_key_slot(mbedtls_svc_key_id_t key_id, psa_key_lifetime_t *lifetime,
+					  psa_drv_slot_number_t *slot_number)
+{
+	sicr_key key = find_key(MBEDTLS_SVC_KEY_ID_GET_KEY_ID(key_id));
+
+	uint32_t domain = PLATFORM_KEY_GET_DOMAIN(MBEDTLS_SVC_KEY_ID_GET_KEY_ID(key_id));
+
+	if (domain != MBEDTLS_SVC_KEY_ID_GET_OWNER_ID(key_id) &&
+	    MBEDTLS_SVC_KEY_ID_GET_OWNER_ID(key_id) != 0) {
+		return PSA_ERROR_NOT_PERMITTED;
+	}
+
+	if (key.bits > 0) {
+		*slot_number = MBEDTLS_SVC_KEY_ID_GET_KEY_ID(key_id);
+		*lifetime = PSA_KEY_LIFETIME_FROM_PERSISTENCE_AND_LOCATION(
+			PSA_KEY_PERSISTENCE_READ_ONLY, PSA_KEY_LOCATION_CRACEN);
+		return PSA_SUCCESS;
+	}
+
+	return PSA_ERROR_DOES_NOT_EXIST;
+}
+
+psa_status_t cracen_platform_keys_provision(psa_key_attributes_t *attributes, uint8_t *key_buffer,
+					    size_t key_buffer_size)
+{
+	uint32_t key_id = MBEDTLS_SVC_KEY_ID_GET_KEY_ID(psa_get_key_id(attributes));
+	sicr_key key = find_key(key_id);
+	uint32_t domain = PLATFORM_KEY_GET_DOMAIN(key_id);
+
+	psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+
+	if (key.bits != UINT16_MAX) {
```

Comment:
should UINT16_MAX be a reserved value (its own self-describing macro)

Was zero size considered as "invalid"? They do this for a lot of other things in PSA crypto (handles, key IDs)